domain/artifact.go

```diff
@@ -14,6 +14,12 @@ type Artifact struct {
 	Artifact string
 }
 
+// Clone returns a copy of the Artifact
+func (a Artifact) Clone() Artifact {
+	newArtifact := a
```

Comment:
Nit: I think `return a` here is equivalent as no pointers involved.

Reply:
good point, it's a leftover from previous commit where I thought I might need to do more to fully clone.
I'll update.